nav.go

```diff
@@ -365,6 +345,9 @@ func (dir *dir) sort() {
 		}
 	}
 
+	// Finally sort after filtering it all
+	sort.SliceStable(dir.files, lessfun)
+
```

Comment:
I haven't thought this through, but does it make sense to do all the filtering (i.e. `dironly`, `hidden`, `filter`) first before doing any sorting? It would make the logic cleaner if all of the filtering could be moved to the top of this function, but I don't know if it will cause any issues.

Since you are pretty much rewriting the entire function, it's probably a good time to clean it up as much as possible.